evalml/pipelines/components/transformers/preprocessing/time_series.py

```diff
@@ -56,16 +62,31 @@ def transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed X.
         """
+        if X is None:
+            X = pd.DataFrame()
         # Normalize the data into pandas objects
-        if not isinstance(X, pd.DataFrame):
-            X = pd.DataFrame(X)
-        if y is not None and not isinstance(y, pd.Series):
-            y = pd.Series(y)
-
-        if self.delay_features and not X.empty:
-            X = X.assign(**{f"{col}_delay_{t}": X[col].shift(t)
-                            for t in range(1, self.max_delay + 1)
-                            for col in X})
+        X = _convert_to_woodwork_structure(X)
+        if y is not None:
+            y = _convert_to_woodwork_structure(y)
+
+        if y.logical_type == logical_types.Categorical:
+            original_y = _convert_woodwork_types_wrapper(y.to_series())
+            y_encoded = LabelEncoder().fit_transform(original_y)
+            y = pd.Series(y_encoded, index=original_y.index)
+        else:
+            y = _convert_woodwork_types_wrapper(y.to_series())
+
+        categorical_columns = {name for name, column in X.columns.items() if
```

Comment:
So if we were using straight up pandas, we'd do something like `df.select_dtypes("object")`.  Does woodwork not have anything like that?

Reply:
There is a [select method](https://woodwork.alteryx.com/en/stable/generated/woodwork.datatable.DataTable.select.html#woodwork.datatable.DataTable.select) we can use! This would pair nicely with your suggestion to apply all of the encoding at once.

Reply:
I decided to not go with select because ultimately we just need the column names for categorical columns. Using select returns a datatable so we'd need an extra step to return the column names. I figure this is more direct.